### ISSUE TITLE
BUG: mpl2nc breaks with negative values when creating netcdf variables

### ISSUE THREAD

**Issue by zssherman** [peterkuma/mpl2nc#2]
Hello,

Currently we are using mpl2nc but noticed recently we were getting empty netcdf files. I was trying to pinpoint the issue, but it seems to be related to a the new numpy and how datatypes are handled.

https://github.com/numpy/numpy/issues/25639

The specific error i am getting with mpl2nc is
```
var = f.createVariable(k, NC_TYPE[h['dtype']], h['dims'],
                                     fill_value=FILL_VALUE[h['dtype']])
OverflowError: Python integer -999 out of bounds for uint16
```
in mpl2nc.py

I think prior if negative values were present, numpy would do value-based promotion, which is no longer the case. I don't believe uint handles negative values, so I was trying to think of a workaround or if there was an idea on how to go about a fix.

My temporary probably not clean fix was:
```
if NC_TYPE[h['dtype']] == 'u2' and FILL_VALUE[h['dtype']] < 0:
    var = f.createVariable(k, 'int16', h['dims'],
        fill_value=FILL_VALUE[h['dtype']])
else:
    var = f.createVariable(k, NC_TYPE[h['dtype']], h['dims'],
        fill_value=FILL_VALUE[h['dtype']])
```
Or unless it can be solved here by changing the fill value for uint16 to a positive number:
```
FILL_VALUE = {
    'float32': -999.0,
    'float64': -999.0,
    'int8': 255,
    'int16': -999,
    'int32': 2147483647,
    'S19': '',
    'uint8': 255,
    'uint16': -999, # this line here
    'uint32': 4294939996,
    'uint64': 18446744073709551615,
}
```

**Comment by peterkuma** [comment]
Hi Zach,

Thank you very much for reporting this issue. I have fixed it in version 1.4.2, now available on PyPI.

Cheers,

Peter

**Comment by zssherman** [comment]
@peterkuma Perfect! Thank you!
